### PR TITLE
Updated technical documents to use the new Layer protocol.

### PR DESCRIPTION
**Pull request by QuantumTCode**
Replaced instances of `applied(to:)` to `call(_:)`.

### CODE REVIEW

File: docs/DifferentiableTypes.md [hunk 1]
```diff
@@ -77,14 +77,14 @@ struct DenseLayer: Differentiable {
...
-    func applied(to input: Tensor<Float>) -> Tensor<Float> {
+    func call(_ input: Tensor<Float>) -> Tensor<Float> {
         return matmul(input, weight) + bias
     }
 }
 
 // Differential operators like `gradient(at:in:)` just work!
 let dense = DenseLayer(weight: [[1, 1], [1, 1]], bias: [0, 0])
-let 𝛁dense = gradient(at: dense) { dense in dense.applied(to: [[3, 3]]).sum() }
+let 𝛁dense = gradient(at: dense) { dense in dense.call(_: [[3, 3]]).sum() }
```
Comment (rxwei) [inline docs/DifferentiableTypes.md]:
```suggestion
let 𝛁dense = gradient(at: dense) { dense in dense([[3, 3]]).sum() }
```

**Comment by QuantumTCode** [comment]
I signed it!

File: docs/ParameterOptimization.md [hunk 2]
```diff
@@ -20,15 +20,15 @@ struct MyMLModel {
...
+    func call(_ input: Tensor<Float>) {
         let h = relu(input • weight1 + bias1)
         return sigmoid(h • weight2 + bias2)
     }
 }
 
 let model = MyMLModel(...)
 let input = Tensor<Float>([0.2, 0.4])
-print(model.applied(to: input))
+print(model.call(_: input))
```
Comment (rxwei) [inline docs/ParameterOptimization.md]:
```suggestion
print(model(input))
```

File: docs/ParameterOptimization.md [hunk 3]
```diff
@@ -368,7 +368,7 @@ class SGD<Model, Scalar: TensorFlowFloatingPoint>
 // Example optimizer usage.
 var dense = DenseLayer(weight: [[1, 1], [1, 1]], bias: [1, 1])
 let input = Tensor<Float>(ones: [2, 2])
-let 𝛁dense = dense.gradient { dense in dense.applied(to: input) }
+let 𝛁dense = dense.gradient { dense in dense.call(_: input) }
```
Comment (rxwei) [inline docs/ParameterOptimization.md]:
```suggestion
let 𝛁dense = dense.gradient { dense in dense(input) }
```

File: docs/ParameterOptimization.md [hunk 4]
```diff
@@ -472,9 +472,9 @@ struct Classifier: Layer {
         l2 = Dense<Float>(inputSize: hiddenSize, outputSize: 1, activation: relu)
     }
     @differentiable
-    func applied(to input: Tensor<Float>) -> Tensor<Float> {
-        let h1 = l1.applied(to: input)
-        return l2.applied(to: h1)
+    func call(_ input: Tensor<Float>) -> Tensor<Float> {
+        let h1 = l1.call(_: input)
+        return l2.call(_: h1)
```
Comment (rxwei) [inline docs/ParameterOptimization.md]:
```suggestion
        return l2(h1)
```

File: docs/ParameterOptimization.md [hunk 5]
```diff
@@ -472,9 +472,9 @@ struct Classifier: Layer {
         l2 = Dense<Float>(inputSize: hiddenSize, outputSize: 1, activation: relu)
     }
     @differentiable
-    func applied(to input: Tensor<Float>) -> Tensor<Float> {
-        let h1 = l1.applied(to: input)
-        return l2.applied(to: h1)
+    func call(_ input: Tensor<Float>) -> Tensor<Float> {
+        let h1 = l1.call(_: input)
```
Comment (rxwei) [inline docs/ParameterOptimization.md]:
```suggestion
        let h1 = l1(input)
```

File: docs/ParameterOptimization.md [hunk 6]
```diff
@@ -484,7 +484,7 @@ let y: Tensor<Float> = [[0], [1], [1], [0]]
 
 for _ in 0..<3000 {
     let 𝛁model = classifier.gradient { classifier -> Tensor<Float> in
-        let ŷ = classifier.applied(to: x)
+        let ŷ = classifier.call(_: x)
```
Comment (rxwei) [inline docs/ParameterOptimization.md]:
```suggestion
        let ŷ = classifier(x)
```

File: docs/DifferentiableFunctions.md [hunk 7]
```diff
@@ -262,7 +262,7 @@ extension Layer {
             backpropagator: (_ direction: Output.CotangentVector)
                 -> (layerGradient: CotangentVector, inputGradient: Input.CotangentVector)) {
         let (out, pullback) = valueWithPullback(at: input) { layer, input in
-            return layer.applied(to: input)
+            return layer.call(_: input)
```
Comment (rxwei) [inline docs/DifferentiableFunctions.md]:
```suggestion
            return layer(input)
```

**Comment by QuantumTCode** [comment]
Ok, got it!

**Pull request merged**